src/js/components/Grommet/Grommet.js

```diff
@@ -150,10 +150,16 @@ const Grommet = forwardRef((props, ref) => {
     if (grommetRef.current) setRoots([grommetRef.current]);
   }, [grommetRef]);
 
+  const rootsContextValue = useMemo(() => ({ roots, setRoots }), [roots]);
+
+  // TO DO this global roots approach causes extra re-renders at
+  // this level which isn't ideal given it's global
+  console.log(roots);
+
   return (
     <ThemeContext.Provider value={theme}>
       <ResponsiveContext.Provider value={responsive}>
-        <RootsContext.Provider value={roots}>
+        <RootsContext.Provider value={rootsContextValue}>
```

Comment:
If we are outside of the grommet container how do we track the roots?

Reply:
this is a really good point!

Reply:
Contexts can have a default value. Can we make the root context have a default value based on `document`?

Reply:
Good point. Need to think about where the RootsContext Provider lives/how that would work. To clarify, Jessica you're meaning a scenario like this:
```
<Grommet>
...
</Grommet>
<Drop>
// I am outside of Grommet wrapper so I don't have access to RootsContext.Provider from Grommet
</Drop>

Reply:
^ Yes that type of scenario

Reply:
Maybe explore the context to be a ref instead of an array.

Have the event listener look at the ref to decide if it's the current container.